tools/upgradetool/src/main/java/org/openhab/core/tools/internal/YamlConfigurationV1TagsUpgrader.java

```diff
@@ -0,0 +1,195 @@
+                .disable(YAMLGenerator.Feature.WRITE_DOC_START_MARKER) // omit "---" at file start
+                .disable(YAMLGenerator.Feature.SPLIT_LINES) // do not split long lines
+                .enable(YAMLGenerator.Feature.INDENT_ARRAYS_WITH_INDICATOR) // indent arrays
+                .enable(YAMLGenerator.Feature.MINIMIZE_QUOTES) // use quotes only where necessary
+                .enable(YAMLParser.Feature.PARSE_BOOLEAN_LIKE_WORDS_AS_STRINGS).build(); // do not parse ON/OFF/... as
+                                                                                         // booleans
+        objectMapper = new ObjectMapper(yamlFactory);
+        objectMapper.findAndRegisterModules();
+        objectMapper.setVisibility(PropertyAccessor.ALL, JsonAutoDetect.Visibility.NONE);
+        objectMapper.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
+        objectMapper.setSerializationInclusion(Include.NON_NULL);
+        objectMapper.enable(JsonGenerator.Feature.WRITE_BIGDECIMAL_AS_PLAIN);
+    }
+
+    @Override
+    public String getName() {
+        return "yamlTagsListToMap";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Upgrade YAML 'tags' list to map format on V1 configuration files";
+    }
+
+    @Override
+    public boolean execute(String userdataDir, String confDir) {
+        String confEnv = System.getenv("OPENHAB_CONF");
+        // If confDir is set to OPENHAB_CONF, look inside /tags/ subdirectory
+        // otherwise use the given confDir as is
+        if (confEnv != null && !confEnv.isBlank() && confEnv.equals(confDir)) {
+            confDir = Path.of(confEnv, "tags").toString();
+        }
+        Path configPath = Path.of(confDir).toAbsolutePath();
+        logger.info("Upgrading YAML tags configurations in '{}'", configPath);
+
+        try {
+            Files.walkFileTree(configPath, new SimpleFileVisitor<>() {
+                @Override
+                public FileVisitResult visitFile(@NonNullByDefault({}) Path file,
+                        @NonNullByDefault({}) BasicFileAttributes attrs) throws IOException {
+                    if (attrs.isRegularFile()) {
+                        Path relativePath = configPath.relativize(file);
+                        String modelName = relativePath.toString();
+                        if (!relativePath.startsWith("automation") && modelName.endsWith(".yaml")) {
+                            logger.info("Checking {}", file);
+                            convertTagsListToMap(file);
+                        }
+                    }
+                    return FileVisitResult.CONTINUE;
+                }
+
+                @Override
+                public FileVisitResult visitFileFailed(@NonNullByDefault({}) Path file,
+                        @NonNullByDefault({}) IOException exc) throws IOException {
+                    logger.warn("Failed to process {}: {}", file.toAbsolutePath(), exc.getClass().getSimpleName());
```

Comment:
Why the class name and not the exception message ?

Reply:
I was a copy paste from https://github.com/openhab/openhab-core/blob/22db52ecc4badb0758dc07a266b631cd2b616282/bundles/org.openhab.core.model.yaml/src/main/java/org/openhab/core/model/yaml/internal/YamlModelRepositoryImpl.java#L133

I've changed it here to getMessage()

Reply:
You changed in YAML model repository not in the upgrade tool.
Please provide the fix in the YAML model repository not this PR which is centered on the upgrade tool.

Reply:
You just made the change at the wrong place ;)
For the model repository, you provided the change in another of your PR.

Reply:
@jimtng : can you please make the change in upgrade tool and not model repository.

I plan to test the tool this week.

Reply:
oops sorry I didn't see your previous message. I've fixed it now.

Reply:
Thank you